docs/companion.md

```diff
@@ -3,3 +3,683 @@ sidebar_position: 4
+```
+
+:::caution
+Omitting the `secret` in the standalone version will generate a secret for you,
+using the above `crypto` string. But when integrating with Express you must provide it yourself.
+This is an essential security measure.
+:::
+
+#### `uploadUrls` `COMPANION_UPLOAD_URLS`
+
+An allowlist (array) of strings (exact URLs) or regular expressions. Companion will only accept uploads to these URLs. This ensures that your Companion instance is only allowed to upload to your trusted servers and prevents [SSRF](https://en.wikipedia.org/wiki/Server-side_request_forgery) attacks. 
+
+#### `COMPANION_PORT`
+
+The port on which to start the standalone server, defaults to 3020.
+
+#### `redisUrl` `COMPANION_REDIS_URL`
+
+TODO: needs a great explanation if it actually works, so let's figure out if it works
+
+URL to running Redis server.
+If this is set, the state of uploads would be stored temporarily.
+This helps for resumed uploads after a browser crash from the client.
+The stored upload would be sent back to the client on reconnection.
+
+#### `redisOptions`
+
+An object of [options supported by redis client](https://www.npmjs.com/package/redis#options-object-properties).
+This option can be used in place of `redisUrl`.
+
+#### `redisPubSubScope`
+
+Use a scope for the companion events at the Redis server.
+Setting this option will prefix all events with the name provided and a colon.
+
+#### `server`
+
+Configuation options for the underlying server.
+
+| Key / Env var                            | Value             | Description                                                                                                                                                                                                                                                                                                                                                                                                                                    |
+| ---------------------------------------- | ----------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `protocol` `COMPANION_PROTOCOL`          | `http` or `https` | Used to build a URL to reference the Companion instance itself, which is used for headers and cookies. Companion does not serve SSL certificates, so by default you should use `http`. You want to set this to `https` if you’ve enabled SSL/HTTPS for your domain by running a reverse https-proxy in front of Companion, or with a built-in HTTPS feature of your hosting service, such as Heroku.                                     |
```

Comment:
```suggestion
| `protocol` `COMPANION_PROTOCOL`          | `http` or `https` | Used to build a URL to reference the Companion instance itself, which is used for headers and cookies. Companion only supports running a HTTP server, however in production HTTPS is usually expected, so you would want to set this to `https` when running in production. You then need to set up a reverse proxy with SSL termination in front of Companion. Many cloud providers have built-in HTTPS termination.                                     |
```

because someone was confused by this here: https://community.transloadit.com/t/companion-generates-redirect-urls-without-ssl/16264

Reply:
oh, but i'm not sure if we should recommend using `http`, because from a security standpoint, using http without SSL is strongly discouraged, when on the internet. Also not sure if we should recommend Heroku and give them free publicity

Reply:
Good points! Updated the comment